### PR TITLE
Fix text widget dirty state calculation w/r/t localized placeholder #10873

### LINKED ISSUE #10873: Card/widget dirty state is not resolvable after discarding card/widget changes in Graph Designer
This results in un-dismissable buttons that only appear when the Graph Designer component has a dirty state. This appears to be related to i18n changes.

Steps to reproduce:

1. Create a new Graph
2. Add a node to the graph ( eg. `string` )
3. Navigate to the cards tab
4. Make a change to the Widget label or Card Title
5. Attempt to discard the changes via the `Discard Card Edits` button
6. Notice the changes are discarded, but the dirty state is not resolved

Diff of card and `._card`:

card:
```
{
    "cards": [],
    "widgets": [
        {
            "card_id": "04626822-08ca-11ef-8d26-acde48001122",
            "config": {
                "en": "",
                "width": "100%",
                "maxLength": null,
                "uneditable": false,
                "placeholder": "Enter text",
                "defaultValue": "",
                "i18n_properties": [
                    "placeholder"
                ],
                "label": ""
            },
            "id": "046269bc-08ca-11ef-8d26-acde48001122",
            "label": "",
            "node_id": "04626660-08ca-11ef-8d26-acde48001122",
            "sortorder": null,
            "source_identifier_id": "8924711c-3dae-44d2-a5cd-2513d9f78159",
            "visible": true,
            "widget_id": "10000000-0000-0000-0000-000000000001",
            "disabled": false
        }
    ],
    "tiles": [],
    "cardid": "04626822-08ca-11ef-8d26-acde48001122",
    "nodegroup_id": "04626660-08ca-11ef-8d26-acde48001122",
    "cardinality": "1",
    "name": "string",
    "instructions": "",
    "cssclass": null,
    "helptext": "",
    "helpenabled": false,
    "helptitle": "",
    "visible": true,
    "active": true,
    "sortorder": null,
    "component_id": "f05e4d3a-53c1-11e8-b0ea-784f435179ea",
    "config": null,
    "constraints": [
        {
            "uniquetoallinstances": false,
            "nodes": [],
            "cardid": "04626822-08ca-11ef-8d26-acde48001122",
            "constraintid": "92e1e7c5-90f2-4cab-ae0a-f01e5181199a"
        }
    ],
    "id": "04626822-08ca-11ef-8d26-acde48001122",
    "description": null,
    "graph_id": "0462653e-08ca-11ef-8d26-acde48001122",
    "source_identifier_id": "fdc5f29d-9a5a-4dae-9819-b4d3c506c166",
    "nodegroup": {
        "cardinality": "1",
        "legacygroupid": null,
        "nodegroupid": "04626660-08ca-11ef-8d26-acde48001122",
        "parentnodegroup_id": null
    }
}
```
._card:
```
{
    "cards": [],
    "widgets": [
        {
            "card_id": "04626822-08ca-11ef-8d26-acde48001122",
            "config": {
                "en": "",
                "width": "100%",
                "maxLength": null,
                "uneditable": false,
                "placeholder": {
                    "en": "Enter text"
                },
                "defaultValue": {
                    "en": {
                        "value": "",
                        "direction": "ltr"
                    }
                },
                "i18n_properties": [
                    "placeholder"
                ],
                "label": ""
            },
            "id": "046269bc-08ca-11ef-8d26-acde48001122",
            "label": "",
            "node_id": "04626660-08ca-11ef-8d26-acde48001122",
            "sortorder": null,
            "source_identifier_id": "8924711c-3dae-44d2-a5cd-2513d9f78159",
            "visible": true,
            "widget_id": "10000000-0000-0000-0000-000000000001",
            "disabled": false
        }
    ],
    "tiles": [],
    "cardid": "04626822-08ca-11ef-8d26-acde48001122",
    "nodegroup_id": "04626660-08ca-11ef-8d26-acde48001122",
    "cardinality": "1",
    "name": "string",
    "instructions": "",
    "cssclass": null,
    "helptext": "",
    "helpenabled": false,
    "helptitle": "",
    "visible": true,
    "active": true,
    "sortorder": null,
    "component_id": "f05e4d3a-53c1-11e8-b0ea-784f435179ea",
    "config": null,
    "constraints": [
        {
            "uniquetoallinstances": false,
            "nodes": [],
            "cardid": "04626822-08ca-11ef-8d26-acde48001122",
            "constraintid": "92e1e7c5-90f2-4cab-ae0a-f01e5181199a"
        }
    ],
    "id": "04626822-08ca-11ef-8d26-acde48001122",
    "description": null,
    "graph_id": "0462653e-08ca-11ef-8d26-acde48001122",
    "source_identifier_id": "fdc5f29d-9a5a-4dae-9819-b4d3c506c166",
    "nodegroup": {
        "cardinality": "1",
        "legacygroupid": null,
        "nodegroupid": "04626660-08ca-11ef-8d26-acde48001122",
        "parentnodegroup_id": null
    }
}
```

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
When a fresh string node is created in the graph designer, the Discard Card Edits button was malfunctioning. Now, the situation is improved, although there are still some related bugs.

More specifically:

    These issues are fixed for a new widget that has
    *just* been created in the graph designer.
    - after discarding edits to fields like widget name,
      there is no dirty state (good).
    - dirty state is correctly picked up for initial
      edits to placeholder.
    
    This issue remains:
    - after dismissing card edits, the placeholder
      field is not editable (dirty state not detected)
      and neither is max length (shows "[object Object]")

### Issues Solved
Closes #10873

### Checklist

-   I targeted one of these branches:
    - [x] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Ticket Background
*   Sponsored by: Getty Conservation Institute
*   Found by: @chrabyrd
*   Tested by: @jacobtylerwalls

### CODE REVIEW

File: arches/app/media/js/views/components/widgets/rich-text.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ define([
         const currentLanguage = {"code": arches.activeLanguage};
         let currentValue = koMapping.toJS(self.value) || initialCurrent;
         self.currentLanguage = ko.observable(currentLanguage);
+        self.currentPlaceholder = ko.observable();
```
Comment (jacobtylerwalls) [inline arches/app/media/js/views/components/widgets/rich-text.js:32]: The crux of this solution is to just treat the placeholder field like a genuine localized field and ensure we're never using shortcuts that immediately unwrap to a certain language.

Comment (jacobtylerwalls) [reply arches/app/media/js/views/components/widgets/rich-text.js:32]: ... er, only use shortcuts in the "presentation layer" of the template without altering card truth, which is why this variable uses "current" in its name

**Comment by jacobtylerwalls** [comment]
The remaining issues are inconvenient but not so serious: you can refresh the page to resolve them.

**Review by chiatt — COMMENTED** [review]
The issue in the graph designer is solved with this fix, but I noticed in the resource editor, this error occurs when rendering a string value: 

image

**Review by chiatt — APPROVED** [review]

**Pull request merged**